src/main/java/org/sonarlint/intellij/config/global/SonarLintGlobalSettings.java

```diff
@@ -47,13 +48,52 @@
   private boolean autoTrigger = true;
   private List<SonarQubeServer> servers = new LinkedList<>();
   private List<String> fileExclusions = new LinkedList<>();
-  private Set<String> includedRules = new HashSet<>();
-  private Set<String> excludedRules = new HashSet<>();
+  @Deprecated
+  private final Set<String> includedRules = new HashSet<>();
+  @Deprecated
+  private final Set<String> excludedRules = new HashSet<>();
+  private final Map<String, Rule> rules = new HashMap<>();
 
   public static SonarLintGlobalSettings getInstance() {
     return ApplicationManager.getApplication().getComponent(SonarLintGlobalSettings.class);
   }
 
+  public void setRuleParam(String ruleKey, String paramName, String paramValue) {
+    if (!rules.containsKey(ruleKey)) {
+      rules.put(ruleKey, new Rule(true));
+    }
+    rules.get(ruleKey).params.put(paramName, paramValue);
+  }
+
+  public String getRuleParam(String ruleKey, String paramName) {
+    if (!rules.containsKey(ruleKey)) {
+      return null;
+    }
+    return rules.get(ruleKey).params.get(paramName);
+  }
+
+  public void enableRule(String ruleKey) {
+    setRuleActive(ruleKey, true);
+  }
+
+  public void disableRule(String ruleKey) {
+    setRuleActive(ruleKey, false);
+  }
+
+  private void setRuleActive(String ruleKey, boolean active) {
+    if (!rules.containsKey(ruleKey)) {
+      rules.put(ruleKey, new Rule(false));
+    }
+    rules.get(ruleKey).isActive = active;
+  }
+
+  public boolean isRuleActive(String ruleKey) {
+    if (!rules.containsKey(ruleKey)) {
+      return true;
```

Comment:
Returning `true` is misleading. When a rule has not explicitely be activated or disactivated, it should be the default activation from the analyzer.
I see two options here:
* returns `Optional<Boolean>`, with `Optional.empty()` meaning there is no explicit state
* change the method to `isRuleExplicitlyDisabled` (since it is used in the quick fix to decide if we should offer to disable the rule)